mlflow/server/js/src/components/MetricsPlotView.js

```diff
@@ -1,171 +1,124 @@
 import React from 'react';
-import {
-  Bar,
-  BarChart,
-  CartesianGrid,
-  Legend,
-  Line,
-  LineChart,
-  Tooltip,
-  XAxis,
-  YAxis,
-  ResponsiveContainer,
-} from 'recharts';
 import Utils from '../utils/Utils';
 import _ from 'lodash';
 import PropTypes from 'prop-types';
-import { X_AXIS_STEP, X_AXIS_RELATIVE, X_AXIS_WALL } from './MetricsPlotControls';
+import { X_AXIS_STEP, X_AXIS_RELATIVE } from './MetricsPlotControls';
 import { CHART_TYPE_BAR } from './MetricsPlotPanel';
+import Plot from 'react-plotly.js';
 
-const COLORS = [
-  '#A3C3D9',
-  '#82ca9d',
-  '#8884d8',
-  '#FF82A9',
-  '#FFC0BE',
-  '#AE76A6',
-  '#993955',
-  '#364958',
-];
+const MAX_RUN_NAME_DISPLAY_LENGTH = 36;
 
 export class MetricsPlotView extends React.Component {
   static propTypes = {
     runUuids: PropTypes.arrayOf(String).isRequired,
+    runDisplayNames: PropTypes.arrayOf(String).isRequired,
     metrics: PropTypes.arrayOf(Object).isRequired,
     xAxis: PropTypes.string.isRequired,
     metricKeys: PropTypes.arrayOf(String).isRequired,
-    showDot: PropTypes.bool.isRequired,
+    // Whether or not to show point markers on the line chart
+    showPoint: PropTypes.bool.isRequired,
     chartType: PropTypes.string.isRequired,
+    isComparing: PropTypes.bool.isRequired,
+    yAxisLogScale: PropTypes.bool.isRequired,
+    lineSmoothness: PropTypes.number,
   };
 
-  static MAX_RUN_NAME_DISPLAY_LENGTH = 36;
-
-  getIdForRunMetricData = (runUuid, metricKey) => `${runUuid}_${metricKey}`;
+  getLineLegend = (metricKey, runDisplayName) => {
+    let legend = metricKey;
+    if (this.props.isComparing) {
+      legend += Utils.truncateString(runDisplayName, MAX_RUN_NAME_DISPLAY_LENGTH);
```

Comment:
Nit: I think you need a colon or space here, otherwise the metric name gets concatenated with the run key

Reply:
Yes. will add in the test commit.